src/components/ReservationForm.tsx

```diff
@@ -0,0 +1,417 @@
+import React, {
+  FunctionComponent,
+  useEffect,
+  useContext,
+  useState,
+} from "react";
+import TextField from "@material-ui/core/TextField";
+import Radio from "@material-ui/core/Radio";
+import RadioGroup from "@material-ui/core/RadioGroup";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import FormControl from "@material-ui/core/FormControl";
+import FormLabel from "@material-ui/core/FormLabel";
+import { CalendarUIProps, CalendarAction } from "../calendar/types";
+import {
+  IconButton,
+  makeStyles,
+  Dialog,
+  Toolbar,
+  Typography,
+  DialogContent,
+  Button,
+} from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
+import Select from "./Select";
+import { AuthContext } from "./AuthContext";
+import { makeTransition } from "./Transition";
+import UserGroup from "../resources/UserGroup";
+import { Formik } from "formik";
+import * as Yup from "yup";
+import EquipmentForm from "./EquipmentForm";
+import QuantityList from "./QuantityList";
+import { ResourceKey } from "../resources/types";
+import Project from "../resources/Project";
+import Equipment from "../resources/Equipment";
+import { quantizeEquipment, buildDictionaries } from "../utils/equipment";
+import { findProjectById } from "../utils/project";
+import {toggleElement} from "../utils/div";
```

Comment:
This totally makes sense, I get it what you're doing with `toggleElement`.
But it seems like you're trying to circumvent the entire React system by doing this.
I need to understand why you aren't using the state and render patterns we are using everywhere else.
```suggestion
```

Reply:
I hadn't even considered a different way to do it, and now that I've thought about it, this way seems ridiculous. Changing it now